lib/health_checks/sidekiq.rb

```diff
@@ -1,68 +1,65 @@
 require 'benchmark'
-require 'socket'
+require 'fileutils'
 require 'health_checks/checks/memory_check'
 require 'health_checks/checks/mongoid_check'
 require 'health_checks/checks/redis_check'
 
 module HealthChecks
   module_function
 
-  def sidekiq(config, mongo_databases, redis_configs)
+  def sidekiq(config, mongo_databases, redis_configs, sleep_seconds: 10)
     config.on(:startup) do
-      LivenessServer.new.start(mongo_databases, redis_configs)
+      LivenessThreadCheck.new.start(mongo_databases, redis_configs, sleep_seconds)
     end
   end
 
   private
 
-  class LivenessServer
-    LIVENESS_PORT = 8080
-
-    def start(mongo_databases, redis_configs)
-      Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
-
-      Thread.start do
-        checks = mongo_databases.map { |db| Checks::MongoidCheck.new(db) }
-        checks += redis_configs.map{ |config| Checks::RedisCheck.new(config) }
-        checks << Checks::MemoryCheck.new
-        server = TCPServer.new(LIVENESS_PORT)
-        loop do
-          Thread.start(server.accept) do |socket|
-            elapsed_time = 0
-            begin
-              checks.each do |check|
-                elapsed_time = Benchmark.measure { check.run }
-                Sidekiq::Logging.logger.info "Time elapsed for #{check} was #{elapsed_time}"
-              end
-              respond_success(socket, 'Live!')
-            rescue => e
-              Sidekiq::Logging.logger.error "Health check failed with #{e.message}"
-              respond_failure(socket, e.message)
-            ensure
-              socket.close
-            end
-          end
+  class LivenessThreadCheck
+    def start(mongo_databases, redis_configs, sleep_seconds)
+      logger = Sidekiq::Logging.logger
+
+      logger.info "Starting liveness thread with #{sleep_seconds} sleep delay"
+
+      checks = mongo_databases.map { |db| Checks::MongoidCheck.new(db) }
+      checks += redis_configs.map{ |config| Checks::RedisCheck.new(config) }
+      # checks << Checks::MemoryCheck.new
```

Comment:
I guess we don't need this because it will be a separate container? If yes, how will we check the memory or we don't need to check the memory anymore(why)?

Reply:
Why was this line removed?

Reply:
Don't commit commented code

Reply:
It should be uncommented. We will still check the memory.